Plugins/Sycl/src/Seeding/CreateSeedsForGroupSycl.cpp

```diff
@@ -0,0 +1,719 @@
+                const auto topSP = deviceTopSPs[deviceIndTop[idx]];
+
+                const auto xM = midSP.x;
+                const auto yM = midSP.y;
+                const auto zM = midSP.z;
+                const auto rM = midSP.r;
+                const auto varianceZM = midSP.varZ;
+                const auto varianceRM = midSP.varR;
+                const auto cosPhiM = xM / rM;
+                const auto sinPhiM = yM / rM;
+
+                const auto deltaX = topSP.x - xM;
+                const auto deltaY = topSP.y - yM;
+                const auto deltaZ = topSP.z - zM;
+
+                const auto x = deltaX * cosPhiM + deltaY * sinPhiM;
+                const auto y = deltaY * cosPhiM - deltaX * sinPhiM;
+                const auto iDeltaR2 = 1.f / (deltaX * deltaX + deltaY * deltaY);
+                const auto iDeltaR = cl::sycl::sqrt(iDeltaR2);
+                const auto cot_theta = deltaZ * iDeltaR;
+
+                detail::DeviceLinEqCircle L;
+                L.cotTheta = cot_theta;
+                L.zo = zM - rM * cot_theta;
+                L.iDeltaR = iDeltaR;
+                L.u = x * iDeltaR2;
+                L.v = y * iDeltaR2;
+                L.er = ((varianceZM + topSP.varZ) +
+                        (cot_theta * cot_theta) * (varianceRM + topSP.varR)) *
+                       iDeltaR2;
+
+                deviceLinTop[idx] = L;
+              }
+            });
+      });
+
+      //************************************************//
+      // **** LINEAR EQUATION TRANSFORMATION - END **** //
+      //************************************************//
+
+      //************************************************//
+      // *********** TRIPLET SEARCH - BEGIN *********** //
+      //************************************************//
+
+      const auto maxMemoryAllocation = std::min(
+          edgesComb, globalBufferSize / uint32_t((sizeof(detail::TripletData) +
+                                                  sizeof(detail::SeedData)) *
+                                                 2));
+
+      detail::TripletData* deviceCurvImpact =
+          cl::sycl::malloc_device<detail::TripletData>(maxMemoryAllocation,
+                                                       (*q));
+
+      // Reserve memory in advance for seed indices and weight
+      // Other way around would allocating it inside the loop
+      // -> less memory usage, but more frequent allocation and deallocation
+      detail::SeedData* deviceSeedArray =
+          cl::sycl::malloc_device<detail::SeedData>(maxMemoryAllocation, *q);
+
+      uint32_t* tripletSearchOffset = cl::sycl::malloc_device<uint32_t>(1, *q);
+
+      // Counting the seeds in the second kernel allows us to copy back the
+      // right number of seeds, and no more.
+      std::atomic_uint32_t* countSeeds =
+          cl::sycl::malloc_device<std::atomic_uint32_t>(1, *q);
+
+      seeds.resize(M);
+      const float MIN = -100000.f;
```

Comment:
I didn't understand how this exactly works, did I understand correctly you could replace it with a bool?
To give you an intuition for the range: the helix curvature is actually stored as 1/diameter, so it's going to be between -1 and 1 (actually much smaller as we usually want helix radii >500mm). 
The maximum allowed impact parameters are configurable and are always >=0

Reply:
I tried that, but I guess too much padding is added to the struct this way, and the extra memory allocations make the code slower.

Reply:
Then never mind, should be a safe value (-1 would also have been small enough). Good to go in from my side!